packages/common/src/types/rollup.ts

```diff
@@ -17,18 +17,25 @@ export enum ERollUpState {
   Failed = 'Failed',
 }
 
-export type TRollUpHistory = TDbRecord<{
+export type TRollUpHistory = {
   databaseName: string;
   currentMerkleTreeRoot: string;
   previousMerkleTreeRoot: string;
   // Previous name `txId` is changed to `transactionObjectId`,
   // txId is not a good name it's alias of tx hash
   transactionObjectId: ObjectId;
   proofObjectId: ObjectId;
+};
+
+// Compound Type
+export type TRollUpTransactionHistory = TRollUpHistory & {
   transactionHash: string;
+  createdAt: Date;
   status: ETransactionStatus;
-  error: string;
-}>;
+  error?: string;
```

Comment:
Based type shouldn't have optional modifier

Reply:
Can we have the `error` empty string instead of null?